ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -130,7 +131,15 @@ connections PeerSelectionActions{peerStateActions = PeerStateActions {monitorPee
                                             . foldr
                                                 ((snd .) . KnownPeers.incrementFailCount)
                                                 (knownPeers st)
-                                            $ (Map.keysSet demotedToCold)
+                                            $ (Map.keysSet demotedToCold),
+
+                          -- When promoting a warm peer, it might happen
+                          -- that the connection will break (or one of the
+                          -- established protocols will error).  For that
+                          -- reason we need to adjust 'inProgressPromoteWarm'.
+                          inProgressPromoteWarm
+                                            = inProgressPromoteWarm
+                                                Set.\\ Map.keysSet demotions
```

Comment:
The [activatePeerConnection](https://github.com/input-output-hk/ouroboros-network/blob/coot/connection-manager/ouroboros-network/src/Ouroboros/Network/PeerSelection/PeerStateActions.hs#L753) (from another branch). which runs in this case has a very little room for a failure, other than a protocol failure which would result in `Warm -> Cold` transition.

Using `demotedToCold` instead of `demotions` (which contains both `-> Cold` and `-> Warm`) should be fine as well.